### DISCUSSION TITLE
Chat function

### DISCUSSION THREAD

**Discussion by relaxibus** [ngageoint/MAGE#5] [Q&A] [answered]
Hi, I'm looking for a TAK alternative and came across MAGE. Is there a integrated chat function like in TAK?
Can we develop own plugins for the phone apps?
Is there any dev server running?

**Comment by newmanw** [comment, marked as answer]
The MAGE mobile app strives to keep the UI as intuitive as possible and focus on core geospatial situational awareness features.  Real-time chat is a non-trivial capability that many iOS and Android apps already do well.  We would expect that customers pick the best for their needs.  For example, using your phone's built-in messaging application like Hangouts, or iMessage, or a more custom and controlled application like [RocketChat](https://www.rocket.chat/).

You can [develop plugins](https://github.com/ngageoint/mage-server/tree/master/plugins) for MAGE Server.  If the plugin APIs are not enough to satisfy your customization requirements, MAGE is open source with a flexible license so you are free to take MAGE and modify to suit your needs.

We do not currently run a development server, but running your own server is fairly easy and lightweight.  See the main [README](https://github.com/ngageoint/mage-server/blob/master/README.md) for instructions.

**Comment by relaxibus** [comment]
Hi, thanks for the answer. Are there some user references using MAGE?
Is the server code also running on ARMv8 platforms? We develop a wearable router for local groups with wireless-cellular-satcom capabilities.

**Reply by restjohn** [reply]
Node.js has an official ARMv8 build, so it should be possible.  The server has a few dependencies with native components, so I'm not 100% certain those have published ARMv8 packages, but stripping those and their related functionality out of the core server should not be too difficult.
